test/gtest/uct/uct_p2p_test.cc

```diff
@@ -182,6 +182,13 @@ void uct_p2p_test::test_xfer_multi_mem_type(send_func_t send, size_t min_length,
     /* Trim at 4.1 GB */
     max_length = ucs_min(max_length, (size_t)(4.1 * (double)UCS_GBYTE));
 
+    /* Trim by BAR1 size if relevant */
+    if (mem_type == UCS_MEMORY_TYPE_CUDA ||
+        mem_type == UCS_MEMORY_TYPE_CUDA_MANAGED) {
+        max_length = ucs_min(max_length,
+                (size_t)(0.8 * mem_buffer::get_bar1_free_size()));
```

Comment:
maybe we need to multiply BAR size by 0.4 (not 0.8) since xfer tests have both send and recv buffer?
I wonder if this really worked on a small BAR card?

Reply:
I have not seen failures on my machine with small bar1 but I do have to some other failures around asymetic connectivity.